Modules/@babylonjs/react-native-windows/windows/BabylonReactNative/EngineView.cpp

```diff
@@ -2,173 +2,452 @@
+		_revokerData.SuspendingRevoker = Application::Current().Suspending(winrt::auto_revoke, [weakThis{ get_weak() }](
+			auto const& /*sender*/,
+			auto const& /*args*/)
+		{
+			if (auto trueThis = weakThis.get())
+				trueThis->OnSuspending();
+		});
+
+		_revokerData.ResumingRevoker = Application::Current().Resuming(winrt::auto_revoke, [weakThis{ get_weak() }](
+			auto const& /*sender*/,
+			auto const& /*args*/)
+		{
+			if (auto trueThis = weakThis.get())
+			{
+				if (CoreWindow::GetForCurrentThread().Visible())
+					trueThis->OnResuming();
+			}
+		});
+		
+		auto swapChainPanel = static_cast<SwapChainPanel>(*this);
+		swapChainPanel.CompositionScaleChanged([this, weak_this{ get_weak() }]
+		(SwapChainPanel const& sender, IInspectable const& object)
+		{
+			if (auto trueThis{ weak_this.get() })
+				trueThis->OnCompositionScaleChanged(sender, object);
+		});
+
+		RegisterInput();
+		RegisterRender();
+
+		s_engineView = this;
+	}
+
+	EngineView::~EngineView()
+	{
+		if(_inputLoopWorker)
+			_inputLoopWorker.Cancel();
+
+		if(_renderLoopWorker)
+			_renderLoopWorker.Cancel();
+		
+		s_criticalSection.lock();
+		_dxgiOutput = nullptr;
+		_backBufferPtr = nullptr;
+		_swapChain = nullptr;
+		s_criticalSection.unlock();
+
+		s_engineView = nullptr;
+	}
+
+	void EngineView::RegisterInput()
+	{
+		WorkItemHandler workItemHandler([this](IAsyncAction const& /* action */) mutable
+		{
+			auto deviceTypes = static_cast<CoreInputDeviceTypes>(
+				static_cast<uint32_t>(Windows::UI::Core::CoreInputDeviceTypes::Mouse) |
+				static_cast<uint32_t>(Windows::UI::Core::CoreInputDeviceTypes::Touch) |
+				static_cast<uint32_t>(Windows::UI::Core::CoreInputDeviceTypes::Pen));
+			auto coreInput = CreateCoreIndependentInputSource(deviceTypes);
+
+			auto PointerPressedRevoker = coreInput.PointerPressed(winrt::auto_revoke, { this, &EngineView::OnPointerPressed });
+			auto PointerMovedRevoker = coreInput.PointerMoved(winrt::auto_revoke, { this, &EngineView::OnPointerMoved });
+			auto PointerReleasedRevoker = coreInput.PointerReleased(winrt::auto_revoke, { this, &EngineView::OnPointerReleased });
+
+			coreInput.Dispatcher().ProcessEvents(Windows::UI::Core::CoreProcessEventsOption::ProcessUntilQuit);
+		});
+
+		// TODO: move to std::thread compared to consuming ThreadPool resources once engine lifecycle bugs are addressed and EngineView's destructor can be successfully invoked.
+		_inputLoopWorker = ThreadPool::RunAsync(workItemHandler, WorkItemPriority::High, WorkItemOptions::TimeSliced);
+	}
+
+	void EngineView::RegisterRender()
+	{
+		s_criticalSection.lock();
+		CreateDeviceResources();
+		CreateSizeDependentResources();
+		s_criticalSection.unlock();
+		
+		// Calculate the updated frame and render once per vertical blanking interval
+		WorkItemHandler workItemHandler([this](IAsyncAction const& action) mutable
+		{
+			//{
+			//	critical_section::scoped_lock lock(s_criticalSection);
+			//	Babylon::EnableView();
+			//}
+

```

Comment:
should we file a follow up issue to look into reenabling this commented out code?